### DISCUSSION TITLE
Multi-factor authentication on Mac

### DISCUSSION THREAD

**Discussion by vinamelody** [pnp/powershell#86] [Q&A]
Hi, I come here after finding that MS Powershell Online module doesn't work with mac.

I tried `Connect-pnponline -Url https://mytenant.sharepoint.com`. But my account has MFA activated. How to get around it?

```
Connect-PnPOnline: AADSTS50076: Due to a configuration change made by your administrator, or because you moved to a new location, you must use multi-factor authentication to access '00000003-0000-0ff1-ce00-000000000000'.
Trace ID: 8ed0b5f0-4456-4e9a-bf1e-fefd1334fe00
Correlation ID: aec480c6-8a04-402f-b13c-10e19a321896
Timestamp: 2021-01-06 15:41:00Z
```

Thanks!

**Comment by ToddKlindt** [comment]
Try `Connect-pnponline -Url https://mytenant.sharepoint.com -UseWebLogin` and see if that gets you connected.
What kind of MFA does your tenant use? Is it the Azure MFA or a 3rd party?

**Reply by vinamelody** [reply]
because i'm using mac, i get `Connect-PnPOnline: -UseWebLogin only works when running on Microsoft Windows due to the requirement to show a login window.`

we use Microsoft MFA (phone call)
is there other alternative?

**Comment by veronicageek** [comment]
Try `Connect-PnPOnline -Url <yourUrl> -PnPManagementShell`. It worked when I tested with an MFA user on a Mac.

**Reply by vinamelody** [reply]
On Mac, using VSCode or Terminal, this also showing me this screen. But if I am using Windows, i don't see it.

<img width="513" alt="Screenshot 2021-03-22 at 4 07 54 PM" src="https://user-images.githubusercontent.com/990511/111975723-e23e5800-8b3b-11eb-8da2-63e774b14be3.png">

**Reply by ToddKlindt** [reply]
What is the full command you're executing when you get this?

**Reply by vinamelody** [reply]
Hi @ToddKlindt 

On Mac, using iTerm (same as VSCode with Powershell extension installed)

```
pwsh
Install-Module -Name "PnP.PowerShell"
Connect-PnPOnline -Url https://mytenant.sharepoint.com -Interactive
```

I've tried `Connect-PnPOnline -Url https://mytenant.sharepoint.com/sites/teamsite -Interactive` too

I use Google Chrome with an already logged in AD account. I created and Site Owner of that `teamsite`

Ironically, my contractor who's using windows, using Powershell ISE, I added him as a Site Owner, and he can connect to that `teamsite`

**Comment by evlo** [comment]
`-Interactive` works best in my experience, if it is possible in your scenario

**Reply by djust270** [reply]
I seem to run into an issue when using -UseWebLogin or -Interactive. It seems to be using some cached credentials. The modern auth windows will open then close immediately.  Is there a way around this? For now I just run it inside Windows Sandbox.

**Reply by ToddKlindt** [reply]
Try the `-ForceAuthentication` parameter.

**Reply by djust270** [reply]
Thank you, that works beautifully `-interactive` with `-ForceAuthentication`

**Reply by vinamelody** [reply]
Using `-Interactive` triggers web login but I experience different behavior using the same Site Owner user. 

- On Windows, it works immediately after login
- On Mac, I got this screen ... why 😐

<img width="513" alt="Screenshot 2021-03-22 at 4 07 54 PM" src="https://user-images.githubusercontent.com/990511/111963194-32aeb900-8b2e-11eb-9917-2c7ea14bb01c.png">
